utils/src/checks.rs

```diff
@@ -49,3 +51,19 @@ pub fn check_rent_exempt(account: &AccountInfo) -> ProgramResult {
     }
     Ok(())
 }
+
+pub fn check_token_account_owner(
+    account: &AccountInfo,
+    owner: &Pubkey,
+) -> Result<Account, ProgramError> {
+    let token_account = Account::unpack_from_slice(&account.data.borrow())?;
```

Comment:
I think we should add an ownership check to `account` otherwise it seems like a footgun as people might assume it does it

Reply:
I was thinking that it would be better to have to add it explicitly with the other checks for the readability, not sure what is best

Reply:
But can do for sure